ethpm/validation.py

```diff
@@ -46,12 +59,14 @@ def validate_registry_uri(uri: str) -> None:
         parsed.path,
         parsed.query,
     )
-    validate_registry_uri_scheme(scheme)
-
-    validate_registry_uri_authority(authority)
-    if query:
-        validate_registry_uri_version(query)
-    validate_package_name(pkg_name[1:])
+    try:
+        validate_registry_uri_scheme(scheme)
+        validate_registry_uri_authority(authority)
+        if query:
+            validate_registry_uri_version(query)
+        validate_package_name(pkg_name[1:])
+    except ValidationError:
+        raise CannotHandleURI("{0} is not a valid registry URI.".format(uri))
```

Comment:
Hrm, this seems to violate my personal "principle of least surprise".  I expect validation functions to do exactly one of two things.

- return `None`
- raise a `ValidationError`

I'd expect the `CannotHandleURI` exception to occur at a different level in the code.